systest/cluster/nodes.go

```diff
@@ -71,7 +71,7 @@ func deployPoet(ctx *testcontext.Context, name string, gateways []string, flags
 		args = append(args, "--gateway="+gw)
 	}
 	ctx.Log.Debugw("deploying poet pod", "args", args, "image", ctx.PoetImage)
-	labels := nodeLabels("poet")
+	labels := nodeLabels(name)
```

Comment:
💡 These are not Deployments, but a Pod and a Service.

A node is given a list of the hostnames of Kubernetes Services that expose PoET Pods. There are N Services and N PoET pods. We want each Service to expose a different PoET Pod. A Service selects Pod it exposes by using a `label` selector. To be able to select a different PoET Pod in each Service we must use a different label for every Pod.

Previously, every Service exposed all available PoETs because they shared the label `app=poet`. Requests to a Service are load balanced across the matched Pods. It led to a situation when a Node thought it was connecting to `poet-0`, but it was actually being connected randomly to one of the available poets.